docs/api/endpoints.md

```diff
@@ -440,3 +440,205 @@ Status: 200 OK
+    "text": "anything that user has to write",
+}
+```
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "posted_by": "01ade2ff-ab21-231f-a12b3c4d5e79",
+    "posted_on": "twitter",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z",
+    "scheduled_time": "2018-10-01T11:00:00Z",
+    "text": "anything that user has to write",
+    "image": null,
+    "is_approved": false,
+    "approved_time": null,
+    "is_posted": false,
+    "posted_time": null
+
+}
+```
+
+## Update a post
+
+```
+PATCH /api/posts/:post_id (requires authentication)
+```
+
+**Request**
+```json
+{
+    "posted_on": "facebook",
+    "text": "some new content",
+    "image":"http://xyz.com/url/of/uploaded_image.jpg",
+    "scheduled_time": "2018-10-01T11:00:00Z",
+}
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "posted_by": "01ade2ff-ab21-231f-a12b3c4d5e79",
+    "posted_on": "facebook",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-09-03T14:23:01Z",
+    "scheduled_time": "2018-10-01T11:00:00Z",
+    "text": "some new content",
+    "image": "http://xyz.com/url/of/uploaded_image.jpg",
+    "is_approved": false,
+    "approved_time": null,
+    "is_posted": false,
+    "posted_time": null
+}
+```
+
+## Review a post
+
+```
+GET /api/posts/:post_id
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "posted_by": "01ade2ff-ab21-231f-a12b3c4d5e79",
+    "posted_on": "twitter",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-09-03T14:23:01Z",
+    "scheduled_time": "2018-10-01T11:00:00Z",
+    "text": "anything that user has to write",
+    "image": "http://xyz.com/url/of/uploaded_image.jpg",
+    "is_approved": false,
+    "approved_time": null,
+    "is_posted": false,
+    "posted_time": null
+}
+```
+
+## Approving a post
+
+```
+POST /api/posts/:post_id/approve (requires authentication)
+```
+
+__NOTE__:Only an Admin or a Moderator can approve a post.
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "posted_by": "01ade2ff-ab21-231f-a12b3c4d5e79",
+    "posted_on": "twitter",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-09-03T14:23:01Z",
+    "scheduled_time": "2018-10-01T11:00:00Z",
+    "text": "anything that user has to write",
+    "image": "http://xyz.com/url/to/uploaded_image.jpg",
+    "is_approved": true,
+    "approved_time": "2018-09-05T15:20:00Z",
+    "is_posted": false,
+    "posted_time": null
+}
+```
+
+## Deleting a post
+
+```
+DELETE /api/posts/:post_id (requires authentication)
+```
+
+**Response**
+Status: 204 No-Content
+
+## Add an image to the social media post
+
+```
+POST /api/posts/:post_id/upload_image (requires authentication)
+```
+
+Image will be uploaded as multipart data as a streaming HTTP request.
```

Comment:
Add a note mentioning the name of the key in the request which will be attached to the post.